### PR TITLE
[KYUUBI #659] Add UDF engine_name

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _Why are the changes needed?_

Add UDF engine_name to show the runtime information about execution engine

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala [hunk 1]
```diff
@@ -36,6 +37,14 @@ object KDFRegistry {
     "string",
     "1.3.0")
 
+  val engine_name: KyuubiDefinedFunction = create(
+    "engine_name",
+    udf(() => SparkContext.getOrCreate().getConf.get("spark.app.name")).asNonNullable(),
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: SparkSession.active.sparkContext.appName

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala [hunk 2]
```diff
@@ -36,6 +37,14 @@ object KDFRegistry {
...
+    "Return the engine_name of Kyuubi Server",
```
Comment (yaooqinn) [inline externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:43]: Return the  spark application name for the associated query engine

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala (hunk 1 above)
Comment (zhaomin1423) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: Thanks, I will update it later.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala (hunk 2 above)
Comment (zhaomin1423) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:43]: Thansk for the suggestion.

File: externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala (hunk 1 above)
Comment (zhaomin1423) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: 'SparkSession.active' can't get the SparkSession when it is running in the executor, so the 'SparkSession.active.sparkContext.appName' have an exception of 'No active or default Spark session found.'

Comment (turboFei) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: How about `SparkEnv.get.conf`?

Comment (zhaomin1423) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: SparkEnv.get.conf can work well,  what is the difference between SparkEnv.get and SparkContext.getOrCreate().getConf, It looks like that their output is the same about appName.

Comment (turboFei) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: SparkContext only existing in driver.

For sparkEnv, it is about seria/lizerManager/mapOutputTracker/blockManager.
And there are driverEnv and executorEnv.
So, you can access it both on driver and executor.

Comment (turboFei) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: FYI:
https://github.com/apache/spark/blob/554d5fef13579cefea083e69b98bce32c0c99126/core/src/main/scala/org/apache/spark/SparkEnv.scala#L197

Comment (zhaomin1423) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala:42]: Thanks, I will update it later.

Comment (yaooqinn) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala]: can we get `val appName = SparkSession.active.sparkContext.appName` at driver side ahead and then pass  the `appName` into the udf?

Comment (zhaomin1423) [reply externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/udf/KDFRegistry.scala]: Make sense, getting it directly is more concise unless there are some advantages.

**Review by turboFei — APPROVED** [review]
Thanks

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
it conflicts, can you resolve it? @zhaomin1423

**Review by yaooqinn — APPROVED** [review]

File: kyuubi-common/src/test/scala/org/apache/kyuubi/operation/JDBCTests.scala [hunk 3]
```diff
@@ -360,6 +361,14 @@ trait JDBCTests extends BasicJDBCTests {
     }
   }
 
+  test("kyuubi defined function - engine_name") {
```
Comment (pan3793) [inline kyuubi-common/src/test/scala/org/apache/kyuubi/operation/JDBCTests.scala:364]: The test will fail in the MiniKube integration test due to the docker image issue. Please temporarily ignore it just like #938.

**Comment by pan3793** [comment]
I'm going to merge it and send a following PR to ignore the test until we fix the docker issue.

**Pull request closed without merging**